File: ios/AITutor/Domain/Model/Listening.swift
import Foundation

struct ListeningContent: Codable, Equatable {
    let title: String
    let topic: String?
    let transcript: String
    let wordCount: Int
    let duration: String?
    let questions: [ReadingQuestion]

    init(
        title: String,
        topic: String? = nil,
        transcript: String,
        wordCount: Int = 0,
        duration: String? = nil,
        questions: [ReadingQuestion]
    ) {
        self.title = title
        self.topic = topic
        self.transcript = transcript
        self.wordCount = wordCount
        self.duration = duration
        self.questions = questions
    }
}

File: android/app/src/main/java/com/aitutor/app/domain/model/Listening.kt
package com.aitutor.app.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class ListeningContent(
    val title: String,
    val topic: String? = null,
    val transcript: String,
    val wordCount: Int = 0,
    val duration: String? = null,
    val questions: List<ReadingQuestion>
)
